standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDefaultTransformer.java

```diff
@@ -833,9 +833,9 @@ private Table validateTablePaths(Table table) throws MetaException {
         dbLocation = Path.getPathWithoutSchemeAndAuthority(new Path(db.getLocationUri()));
         Path tablePath = null;
         if (!FileUtils.isSubdirectory(whRootPath.toString(), dbLocation.toString())) {
-          tablePath = new Path(db.getLocationUri(), table.getTableName());
+          tablePath = new Path(db.getLocationUri(), table.getTableName().toLowerCase());
```

Comment:
Please check MetaStoreUtils.encodeTableName is required for tableName at here as well.